add display of sonata_help on many form type

nykopol opened the pull request:
I am targeting this branch, because the changes should not break BC.

## Changelog

```markdown
### Fixed
- Add display of missing `sonata_help` on many form type
```

## Subject

With the default form template, on many form type, the `sonata_help` option is not displayed.

This PR solve this issue for the following types : 

- Textarea
- Date
- DateTime
- Number
- Money
- Percent
- Choice
- Boolean

> DatePicker and DateTimePicker are not affected by this PR as they corresponding template are in the core bundle.

### Preview before and after the PR

Example before
![image](https://user-images.githubusercontent.com/2706685/46658631-d6ed5100-cbbb-11e8-90ef-6f7b7ea6f88d.png)

Example after
![image](https://user-images.githubusercontent.com/2706685/46658690-f4221f80-cbbb-11e8-99e9-a52c0222859c.png)

greg0ire commented:
Are there any docs that need to be updated?

nykopol commented:
I don't think so as it's a bug fix.

greg0ire commented:
> I don't think so as it's a bug fix.

That's why I asked lol. If there are no docs, it's a feature, if there are docs, and they promise something that does not exist, it is a bugfix. Can you show the existing docs about this?

OskarStark commented on Resources/views/Form/form_admin_fields.html.twig:
```diff
@@ -263,6 +277,9 @@ file that was distributed with this source code.
         </div>
     {% endif %}
+    {% if datepicker_use_button is not defined %}
+        {{ block('sonata_help') }}
```
is this correct? why only show help if button is not defined?

greg0ire replied:
Indeed, care to explain?

OskarStark replied:
IMO you can remove the new if statement here, will it work @nykopol ?

nykopol replied:
For the `DatePicker` and `DateTimePicker` types I explicitly disable the help message when there is a button (the calendar icon next to the form) because it break the rendering. 

For this two form type, this PR doesn't work as they corresponding template are in `sonata-project/core-bundle`. I'll do the corresponding PR if this one is accepted.

OskarStark commented on Resources/views/Form/form_admin_fields.html.twig:
```diff
@@ -321,6 +339,9 @@ file that was distributed with this source code.
         </div>
     {% endif %}
+    {% if datepicker_use_button is not defined %}
+        {{ block('sonata_help') }}
```
same here?

nykopol replied:
yes

nykopol commented:
@greg0ire about the question if this is a feature or a bugfix, i just checked the documentation. [In the admin documentation](https://sonata-project.org/bundles/admin/master/doc/reference/form_help_message.html#help-messages-in-a-sub-field), there isn't a literal explanation, but the code give as example look obvious to me. The `help` option should be replaced with `sonata_help` in subfields.

In addition, the [doctrineOrmAdmin documentation](https://sonata-project.org/bundles/doctrine-orm-admin/master/doc/reference/form_field_definition.html#advanced-usage-one-to-many) is clear:

```
To render child help messages you must use ‘sonata_help’ instead of ‘help’.
```

So, i think that this is realy a bugfix.

nykopol commented:
@OskarStark I used this conditions as the `date_widget` is used in `core-bundle`  for rendering the `datepicker_widget`, so this condition avoid to break the rendering of 'pickers' widgets. I'm aware that this is not optimal.

I don't know what is the best way to deal with this. The 'pickers' fields are part of the core-bundle. We cannot add the `sonata_help` in the original bundle as this option is defined in `admin-bundle` which is not part of the `core-bundle` dependencies. The solutions that i see are either
- we overwrite the template of this fields in `admin-bundle` to add properly the `sonata_help` option
- there is a decoupling problem and we move this fields from the core to the admin bundle. From my point of view, this options is maybe the cleanest but imply BC break.

If there is an other option or if i misunderstood something, I'm open to discuss.

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

OskarStark commented:
Could you please apply/answer the comments?
Is there anything todo ?

Could you add some Screenshots?

OskarStark commented:
@nykopol I understand the two bundle problem, but for now can we easily merge `sonata_help` for the fields except the 2 datepickers. Can you please add some docs on how you can use the help and add some screenshots to the docs and to the PR header?

I really like this PR and would be happy to get this merged 👍 

Thank you for your time!

nykopol commented:
@OskarStark sorry for my inactivity, I don't have a lot of time but I will try to come back on this this weekend.

nykopol commented:
@OskarStark I rebased the PR and added a screenshot in the docs. About adding some explanation on how to use the `sonata_help` option I think that `form_help_message.rst` is explicit. What should I add ?

OskarStark commented:
IMO this PR is ready to merge 👍

OskarStark commented:
I am thinking of a development pre-setup for Sonata, where we could have one Admin configured like yours (used for the screenshots) etc. 🤔 

Do you think this would make sense? Kind of a sandbox, but with custom admins, doing custom stuff on our codebase, this would make it easier to develop special features?? WDYT?

nykopol commented:
@OskarStark absolutely !!! I almost build one each time I want to test if something doesn't work because of my specific configuration or if it's a real bug. Same thing when I make a PR for the same reason.

jorenvh1 commented:
Which release will this be in?

OskarStark commented:
> Which release will this be in?

after merging it, in the next release 👍
